src/Cassandra/YugaByte/PartitionAwarePolicy.cs

```diff
@@ -0,0 +1,321 @@
+﻿
+// Copyright (c) YugaByte, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using Cassandra.Serialization;
+
+namespace Cassandra.YugaByte
+{
+    public class PartitionAwarePolicy : ILoadBalancingPolicy
+    {
+        private readonly ILoadBalancingPolicy _childPolicy = new DCAwareRoundRobinPolicy(null, int.MaxValue);
```

Comment:
This should not be hard-coded but be passed in in the constructor. See `PartitionAwarePolicy.java` counterpart and `TokenAwarePolicy.cs`.